packages/web3-eth-contract/src/contract.ts

```diff
@@ -399,6 +412,35 @@ export class Contract<Abi extends ContractAbi>
 		return sendTransaction(this, tx, DEFAULT_RETURN_FORMAT);
 	}
 
+	private _contractMethodDeploySend<Options extends PayableCallOptions | NonPayableCallOptions>(
+		abi: AbiFunctionFragment,
+		params: unknown[],
+		options?: Options,
+		contractOptions?: ContractOptions,
+	) {
+		const tx = getSendTxParams({
+			abi,
+			params,
+			options,
+			contractOptions: contractOptions ?? this.options,
+		});
+
+		return sendTransaction(this, tx, DEFAULT_RETURN_FORMAT, {
+			transactionResolver: receipt => {
+				if (receipt.status === '0x0') {
+					throw new Web3ContractError('contract deployment error');
```

Comment:
Its imp to include transaction receipt in error data for easy debugging, instead of only string message.